src/transformers/generation_utils.py

```diff
@@ -955,6 +956,11 @@ def generate(
             raise ValueError(
                 "Diverse beam search cannot be used in sampling mode. Make sure that `do_sample` is set to `False`."
             )
+        if not do_sample and not (temperature is None or math.isclose(temperature, 1)):
```

Comment:
We can't really add this error  - there are lots of configs that have `temperature` set to a value != 1 and for which the user might often want to change between `do_sample=True` and `do_sample=False`